deps/build.jl

```diff
@@ -1,96 +1,4 @@
-const URLS = (
-    "https://www.iana.org/time-zones/repository/tzdata-latest.tar.gz",
-    "ftp://ftp.iana.org/tz/tzdata-latest.tar.gz",  # Unreliable source
-)
-
-# See "ftp://ftp.iana.org/tz/data/Makefile" PRIMARY_YDATA for listing of
-# regions to include. YDATA includes historical zones which we'll ignore.
-const REGIONS = (
-    "africa", "antarctica", "asia", "australasia",
-    "europe", "northamerica", "southamerica",
-    # "pacificnew", "etcetera", "backward",  # Historical zones
-)
-
-isdir(TZDATA_DIR) || mkdir(TZDATA_DIR)
-isdir(COMPILED_DIR) || mkdir(COMPILED_DIR)
-
-info("Downloading latest tz database")
-archive = ""
-for url in URLS
-    try
-        archive = download(url)
-        break
-    catch
-        warn("Failed to download tz database from: $url")
-    end
-end
-isfile(archive) || error("Unable to download tz database")
-
-info("Extracting tz database archive")
-extract(archive, TZDATA_DIR, REGIONS)
-rm(archive)
-
-
-info("Converting tz database into TimeZone data")
-for file in readdir(COMPILED_DIR)
-    rm(joinpath(COMPILED_DIR, file), recursive=true)
-end
-compile(TZDATA_DIR, COMPILED_DIR)
-
-if is_windows()
-    translation_dir = dirname(WIN_TRANSLATION_FILE)
-    isdir(translation_dir) || mkdir(translation_dir)
-
-    info("Downloading Windows to POSIX timezone name XML")
-
-    # Retrieve a mapping of Windows timezone names to Olson timezone names.
-    # Details on the contents of this file can be found at:
-    # http://cldr.unicode.org/development/development-process/design-proposals/extended-windows-olson-zid-mapping
-    xml_source = "http://unicode.org/repos/cldr/trunk/common/supplemental/windowsZones.xml"
-    xml_file = try
-        download(xml_source)
-    catch err
-        warn(err)
-        info("Falling back to cached XML")
-        joinpath(translation_dir, "windowsZones.xml")
-    end
-
-    info("Pre-processing Windows translation")
-
-    # Get the timezone conversions from the file
-    xdoc = parse_file(xml_file)
-    xroot = root(xdoc)
-    windows_zones = find_element(xroot, "windowsZones")
-    map_timezones = find_element(windows_zones, "mapTimezones")
-    map_zones = get_elements_by_tagname(map_timezones, "mapZone")
-
-    # TODO: Eliminate the Etc/* POSIX names here? See Windows section of `localzone`
-
-    # Dictionary to store the windows to timezone conversions
-    translation = Dict{AbstractString,AbstractString}()
-    for map_zone in map_zones
-        # Territory "001" is the global default
-        if attribute(map_zone, "territory") == "001"
-            win_name = attribute(map_zone, "other")
-            posix_name = attribute(map_zone, "type")
-            translation[win_name] = posix_name
-        end
-    end
-
-    open(WIN_TRANSLATION_FILE, "w") do fp
-        serialize(fp, translation)
-    end
-end
-
-info("Successfully processed TimeZone data")
+# ENV variable allows us to only download a single version during CI jobs
+build(get(ENV, "JULIA_TZ_VERSION", "latest"))
```

Comment:
Then make a new package release when there's a new tzdata file available. This has been the cause of repeated test failures over the last few years. Better to stick to something that's known to work correctly, and only release new versions after they've been tested.